grammars/puppet.cson

```diff
@@ -32,79 +41,56 @@
         'captures':
           '1':
             'name': 'storage.modifier.puppet'
-        'end': '(?={)'
+        'end': '{'
         'name': 'meta.definition.class.inherits.puppet'
         'patterns': [
           {
             'match': '\\b((?:[-_A-Za-z0-9".]+::)*[-_A-Za-z0-9".]+)\\b'
-            'name': 'support.type.puppet'
+            'name': 'entity.name.type.class.puppet'
           }
         ]
       }
       {
-        'captures':
-          '1':
-            'name': 'variable.other.puppet'
-          '2':
-            'name': 'punctuation.definition.variable.puppet'
-        'match': '((\\$+)[a-zA-Z_][a-zA-Z0-9_]*)\\s*(?=,|\\))'
-        'name': 'meta.function.argument.no-default.puppet'
-      }
-      {
-        'begin': '((\\$+)[a-zA-Z_][a-zA-Z0-9_]*)(?:\\s*(=)\\s*)\\s*'
-        'captures':
-          '1':
-            'name': 'variable.other.puppet'
-          '2':
-            'name': 'punctuation.definition.variable.puppet'
-          '3':
-            'name': 'keyword.operator.assignment.puppet'
-        'end': '(?=,|\\))'
-        'name': 'meta.function.argument.default.puppet'
-        'patterns': [
-          {
-            'include': '#parameter-default-types'
-          }
-        ]
+        'include': '#defineparameters'
       }
     ]
   }
   {
-    'begin': '^\\s*(define)\\s+([a-zA-Z0-9_:]+)\\s*(\\()'
     'beginCaptures':
       '1':
-        'name': 'storage.type.function.puppet'
-      '2':
-        'name': 'entity.name.function.puppet'
-      '3':
-        'name': 'punctuation.definition.parameters.begin.puppet'
-    'contentName': 'meta.function.arguments.puppet'
-    'end': '\\)'
-    'endCaptures':
-      '1':
-        'name': 'punctuation.definition.parameters.end.puppet'
-    'name': 'meta.function.puppet'
+        'name': 'storage.type.puppet'
+    'begin': '((?:[-_A-Za-z0-9".]+::)*[-_A-Za-z0-9".]+)\\s*{'
+    'name': 'meta.definition.resource.puppet'
+    'end': '}'
     'patterns': [
       {
         'captures':
           '1':
-            'name': 'variable.other.puppet'
-          '2':
-            'name': 'punctuation.definition.variable.puppet'
-        'match': '((\\$+)[a-zA-Z_][a-zA-Z0-9_]*)\\s*(?=,|\\))'
-        'name': 'meta.function.argument.no-default.puppet'
+            'name': 'punctuation.classtitle.puppet'
+        'match': '(:)'
```

Comment:
match should go before captures
Also, since all this is matching is a single colon, you can change it to a zero capture.